File: Programmers/Lv2/의상.js
function solution(clothes) {
  var cabinet = {};

  clothes.forEach((v) => {
    if (v[1] in cabinet) {
      cabinet[v[1]] += 1;
    } else {
      cabinet[v[1]] = 1;
    }
  });

  var answer = 1;
  for (let tmp of Object.values(cabinet)) {
    answer *= tmp + 1;
  }
  answer -= 1;

  return answer;
}

File: Programmers/Lv2/의상.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    unordered_map<string, vector<string>> closet;
    
    for(auto& cloth : clothes){
        closet[cloth[1]].push_back(cloth[0]);
    }
    
    int res = 1;
    for(auto& pair : closet){
        string key = pair.first;
        res *= (closet[key].size()+1);
    }
    res -= 1;
    return res;
}